plugins/woocommerce/src/Internal/ReceiptRendering/ReceiptRenderingRestController.php

```diff
@@ -0,0 +1,203 @@
+			gmdate( 'Y-m-d', strtotime( "+{$request->get_param('expiration_days')} days" ) );
+
+		$order_id = $request->get_param( 'id' );
+
+		$filename = wc_get_container()->get( ReceiptRenderingEngine::class )->generate_receipt( $order_id, $expiration_date, $request->get_param( 'force_new' ) );
+
+		return is_null( $filename ) ?
+			new WP_Error( 'woocommerce_rest_not_found', __( 'Order not found', 'woocommerce' ), array( 'status' => 404 ) ) :
+			$this->get_response_for_file( $filename );
+	}
+
+	/**
+	 * Formats the response for both the GET and POST endpoints.
+	 *
+	 * @param string $filename The filename to return the information for.
+	 * @return array The data for the actual response to be returned.
+	 */
+	private function get_response_for_file( string $filename ): array {
+		$expiration_date = TransientFilesEngine::get_expiration_date( $filename );
+		$public_url      = wc_get_container()->get( TransientFilesEngine::class )->get_public_url( $filename );
+
+		return array(
+			'receipt_url'     => $public_url,
+			'expiration_date' => $expiration_date,
+		);
+	}
+
+	/**
+	 * Get the accepted arguments for the GET request.
+	 *
+	 * @return array[] The accepted arguments for the GET request.
+	 */
+	private function get_args_for_get_order_receipt(): array {
+		return array(
+			'id' => array(
+				'description' => __( 'Unique identifier of the order.', 'woocommerce' ),
+				'type'        => 'integer',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+		);
+	}
+
+	/**
+	 * Get the schema for both the GET and the POST requests.
+	 *
+	 * @return array[]
+	 */
+	private function get_schema_for_get_and_post_order_receipt(): array {
+		$schema               = $this->get_base_schema();
+		$schema['properties'] = array(
+			'receipt_url'     => array(
+				'description' => __( 'Public url of the receipt.', 'woocommerce' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+			'expiration_date' => array(
+				'description' => __( 'Expiration date of the receipt, formatted as yyyy-mm-dd.', 'woocommerce' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+		);
+
+		return $schema;
+	}
+
+	/**
+	 * Get the accepted arguments for the POST request.
+	 *
+	 * @return array[]
+	 */
+	private function get_args_for_create_order_receipt(): array {
+		return array(
+			'id'              => array(
+				'description' => __( 'Unique identifier of the order.', 'woocommerce' ),
+				'type'        => 'integer',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+			),
+			'expiration_date' => array(
+				'description' => __( 'Expiration date formatted as yyyy-mm-dd.', 'woocommerce' ),
+				'type'        => 'string',
+				'context'     => array( 'view', 'edit' ),
+				'readonly'    => true,
+				'default'     => null,
+			),
+			'expiration_days' => array(
```

Comment:
I guess this is more of a personal opinion, but `expiration_days` is a very similar name to `expiration_date`, but they take entirely different data types. `days_until_expiration` might help differentiate these more.

Reply:
Isn't `days_until_expiration` a bit verbose? And I don't think parameters with similar names but different types is an issue as long as we document them properly. 

Coincidentally (I didn't know before I decided the names), [Amazon S3 uses the same parameter names](https://docs.ansible.com/ansible/latest/collections/community/aws/s3_lifecycle_module.html):

![image](https://github.com/woocommerce/woocommerce/assets/937723/849b984a-bf28-4bf9-85fd-dd990c2ffb21)